### PR TITLE
User auth

### PR DESCRIPTION
**Pull request by niumis**

### CODE REVIEW

File: src/AppBundle/Resources/views/Security/register.html.twig [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+                        <div class="form-group">
+                            {{ form_widget(form.plainPassword.first, {'attr': {'class': 'form-control', 'placeholder': 'Slaptažodis'} }) }}
+                        </div>
+                        <div class="form-group">
+                            {{ form_widget(form.plainPassword.second, {'attr': {'class': 'form-control', 'placeholder': 'Pakartoti slaptažodį'} }) }}
+                        </div>
+
+                        <button type="submit" class="btn">Registruotis</button>
+                        {{ form_end(form) }}
+
```
Comment (zyrukas) [inline src/AppBundle/Resources/views/Security/register.html.twig]: really annoying space :D

File: src/AppBundle/Resources/scss/template.scss [hunk 2]
```diff
@@ -2058,3 +2058,72 @@ input, select, a, .transition,
...
+}
+
+.user-account input {
+  height: 44px;
+  margin-bottom: 30px;
+  font-size: 16px;
+}
+
+.user-account .btn {
+  background-color: #00a651;
```
Comment (zyrukas) [inline src/AppBundle/Resources/scss/template.scss]: spalvos i kintamuosius

File: src/AppBundle/Resources/scss/template.scss [hunk 3]
```diff
@@ -2058,3 +2058,72 @@ input, select, a, .transition,
...
+/* ==========================================================================
+ 	User Page login / Sign up
+ ========================================================================== */
+
+#main.user-page {
+  padding: 60px 0 80px;
+}
+
+.user-account {
+  background-color: #fff;
```
Comment (zyrukas) [inline src/AppBundle/Resources/scss/template.scss]: spalvos i kintamuosius

File: src/AppBundle/Resources/scss/template.scss [hunk 4]
```diff
@@ -2058,3 +2058,72 @@ input, select, a, .transition,
...
+  margin-top:10px;
+}
+
+.user-page .btn-primary {
+  padding:15px 30px;
+  display:block;
+  font-size:18px;
+  border-radius:4px;
+  margin-top:25px;
+  background-color:#0072bc;
```
Comment (zyrukas) [inline src/AppBundle/Resources/scss/template.scss]: spalvos i kintamuosius

File: src/AppBundle/Resources/scss/template.scss [hunk 5]
```diff
@@ -2058,3 +2058,72 @@ input, select, a, .transition,
...
+}
+
+.user-page .btn-primary:hover {
+  background-color:#00a651;
```
Comment (zyrukas) [inline src/AppBundle/Resources/scss/template.scss]: spalvos i kintamuosius

File: src/AppBundle/Resources/scss/template.scss [hunk 6]
```diff
@@ -2058,3 +2058,72 @@ input, select, a, .transition,
...
+.user-account .user-option {
+  overflow:hidden;
+  margin-top:55px;
+}
+
+.user-option .forgot-password {
+  margin-top:10px;
+}
+
+.user-page .btn-primary {
```
Comment (zyrukas) [inline src/AppBundle/Resources/scss/template.scss]: visas blokas neatitinka psr-2

**Comment by DarkerTH** [comment]
+2

File: src/AppBundle/Entity/User.php [hunk 7]
```diff
@@ -0,0 +1,130 @@
...
+
+    /**
+     * @var string
+     *
+     * @ORM\Column(name="role", type="string", length=255)
+     */
+    private $role;
+
+
+    public function __construct()
```
Comment (zyrukas) [inline src/AppBundle/Entity/User.php]: php docs

File: src/AppBundle/Entity/User.php [hunk 8]
```diff
@@ -0,0 +1,130 @@
...
+    {
+        $this->role = 'ROLE_USER';
+    }
+
+    public function getUsername()
```
Comment (zyrukas) [inline src/AppBundle/Entity/User.php]: php docs

File: src/AppBundle/Entity/User.php [hunk 9]
```diff
@@ -0,0 +1,130 @@
...
+    {
+        return null;
+    }
+
+    public function getRoles()
+    {
+        return [$this->role];
+    }
+
+    public function eraseCredentials()
```
Comment (zyrukas) [inline src/AppBundle/Entity/User.php]: why is this function here?

File: app/config/security.yml [hunk 10]
```diff
@@ -32,8 +34,13 @@ security:
...
+            provider: our_db_provider
             # http_basic: ~
             # http://symfony.com/doc/current/book/security.html#a-configuring-how-your-users-will-authenticate
 
             # form_login: ~
             # http://symfony.com/doc/current/cookbook/security/form_login_setup.html
+
+    access_control:
+        - { path: ^/login, roles: IS_AUTHENTICATED_ANONYMOUSLY }
+        - { path: ^/admin, roles: ROLE_ADMIN }
```
Comment (zyrukas) [inline app/config/security.yml]: no newline

**Comment by zyrukas** [comment]
+1 after fixing comments

**Pull request merged**